Change the guidelines about establishing base direction and mention ALM

aphillips opened the pull request:
Fixes #8. Fixes #18.

This addresses both @stevenatkin's comment and @lkimmel's comment.

Note that I added "isolated" to the guideline about providing attributes in markup languages. I am tempted to want to add text to make isolating preferred to embedding. This section may need further review.

I also change the title of the section about strong marks, adding some text ("Strong directional marks:").


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/136.html" title="Last updated on Oct 17, 2024, 3:32 PM UTC (e16baf4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/136/ba0a752...aphillips:e16baf4.html" title="Last updated on Oct 17, 2024, 3:32 PM UTC (e16baf4)">Diff</a>

aphillips committed:
Change the guidelines about establishing base direction.

Fixes #8.

This addresses both @stevenatkin's comment and @lkimmel's comment.

Note that I added "isolated" to the guideline about providing attributes
in markup languages. I am tempted to want to add text to make isolating
preferred to embedding. This section may need further review.

aphillips committed:
Fix #19: Mention ALM

aphillips committed:
Add U+061C (ALM) image

xfq commented on index.html:
```diff
@@ -875,13 +875,13 @@ <h4>Problems with control characters</h4>
-<p>The first point to be clear about is that neither RLM nor LRM establish the base direction for a range of text.&nbsp; They are simply invisible characters with strong directional properties.</p>
+<h4>Strong direction marks: RLM, LRM, and ALM</h4>
+<p>A word about the Unicode characters <span class="codepoint" translate="no"><img alt="RLM" src="images/200F.png"><code class="uname">U+200F RIGHT-TO-LEFT MARK</code></span> (RLM), <span class="codepoint" translate="no"><img alt="LRM" src="images/200E.png"><code class="uname">U+200E LEFT-TO-RIGHT MARK</code></span> (LRM), and <span class="codepoint" translate="no"><img alt="ALM" src="images/061C ...
+<p>The first point to be clear about is that these three characters do not establish the base direction for a range of text.&nbsp; They are simply invisible characters with strong directional properties.</p>
```
```suggestion
<p>The first point to be clear about is that these three characters do not establish the base direction for a range of text. They are simply invisible characters with strong directional properties.</p>
```

xfq commented:
Overall LGTM, but we should probably link to some background information of these characters. We don't seem to have them our website (see https://github.com/w3c/i18n-drafts/issues/602). Maybe somewhere at unicode.org for now?

aphillips committed:
Update index.html

Co-authored-by: Fuqiao Xue <xfq@w3.org>

aphillips commented on index.html:
```diff
@@ -875,13 +875,13 @@ <h4>Problems with control characters</h4>
-<h4>RLM and LRM</h4>
-<p>A word about the Unicode characters <span class="codepoint" translate="no"><img alt="RLM" src="images/200F.png"><code class="uname">U+200F RIGHT-TO-LEFT MARK</code></span> (RLM) and <span class="codepoint" translate="no"><img alt="LRM" src="images/200E.png"><code class="uname">U+200E LEFT-TO-RIGHT MARK</code></span> (LRM) is warranted at this point.</p>
-<p>The first point to be clear about is that neither RLM nor LRM establish the base direction for a range of text.&nbsp; They are simply invisible characters with strong directional properties.</p>
+<h4>Strong direction marks: RLM, LRM, and ALM</h4>
```
```suggestion
<h4>Strong directional formatting characters: RLM, LRM, and ALM</h4>
```

aphillips committed:
Update index.html

aphillips committed:
Add ALM example and explanatory text. Fix some phrasing.

aphillips committed:
Fix typos

aphillips committed:
Improve example

- Fix text to refer to third example
- Make appearance of the text go under the HTML markup to avoid wrapping
  issues
- Use `spilloverExample` style for the rendered text so it is pretty

aphillips committed:
Merge remote-tracking branch 'upstream/gh-pages' into issue-8

xfq commented on index.html:
```diff
@@ -875,15 +875,44 @@ <h4>Problems with control characters</h4>
+<h4>Strong directional formatting characters: RLM, LRM, and ALM</h4>
+<p>A word about the Unicode characters <span class="codepoint" translate="no"><img alt="RLM" src="images/200F.png"><code class="uname">U+200F RIGHT-TO-LEFT MARK</code></span> (RLM), <span class="codepoint" translate="no"><img alt="LRM" src="images/200E.png"><code class="uname">U+200E LEFT-TO-RIGHT MARK</code></span> (LRM), and <span class="codepoint" translate="no"><img alt="ALM" src="images/061C ...
+<p>The first point to be clear about is that these three characters do not establish the base direction for a range of text. They are simply invisible characters with strong directional properties.</p>
+<p>This means that you cannot use RLM for example, to make the text <kbd>W3C</kbd> appear to the left of the Hebrew text in the following example.</p>
+<p>The title is "<span dir="rtl" lang="he">פעילות הבינאום, W3C</span>".</p>
```
When this text is rendered in the browser, "W3C" is already on the left. Should `dir="rtl"` be removed here?
